### PR TITLE
Psycho Shift uses CanBeStatused Functions

### PR DESCRIPTION
**Pull request by ghoulslash**

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -8661,39 +8661,27 @@ static void Cmd_various(void)
...
-            }
             else
             {
                 gBattleCommunication[MULTISTRING_CHOOSER] = 3;
             }
         }
         else if (gBattleMons[gBattlerAttacker].status1 & STATUS1_PSN_ANY)
         {
-            if (targetAbility == ABILITY_IMMUNITY)
+            if (!CanBePoisoned(gBattlerTarget, targetAbility))
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:8678]: Shouldn't this be `if  (!CanBePoisoned(gBattlerAttacker, gBattlerTarget))`? That's what its arguments say.

**Review by AsparagusEduardo — COMMENTED** [review]
Could we take this chance to fix the messages?
When it can't be statused due to ability, it doesn't show the target's ability for burn, and it doesn't show the target's name for all 3 I tested (burn, poison, paralysis).
Not only that, when the target can't be statused due to type, it erroneously shows the message as if it was the ability protecting it.
(this is happening both before and after the change in the PR)
psycho1psychoburn2

**Comment by ghoulslash** [comment]
Updated to make psycho shift fail if the target cannot acquire the user's status (as per [this conversation](https://discord.com/channels/419213663107416084/774393519569502268/941041926784839710))

ready for review

**Comment by AsparagusEduardo** [comment]
> Updated to make psycho shift fail if the target cannot acquire the user's status (as per [this conversation](https://discord.com/channels/419213663107416084/774393519569502268/941041926784839710))
> 
> ready for review

Adding the conversation for archival purposes:
image

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**